src/Build/Evaluation/SimpleProjectRootElementCache.cs

```diff
@@ -63,6 +63,9 @@ private ProjectRootElement GetFromOrAddToCache(string projectFile, OpenProjectRo
                 ErrorUtilities.VerifyThrowInternalNull(rootElement, "projectRootElement");
                 ErrorUtilities.VerifyThrow(rootElement.FullPath.Equals(key, StringComparison.OrdinalIgnoreCase),
                     "Got project back with incorrect path");
+
+                AddEntry(rootElement);
+
                 ErrorUtilities.VerifyThrow(_cache.TryGetValue(key, out _),
                     "Open should have renamed into cache and boosted");
```

Comment:
This message is now outdated. I'd recommend something like "ProjectRootElement should have been added into cache"